### PR TITLE
ctp: remove final hh tests

### PR DESCRIPTION
**Pull request by tynes**
**Description**

This commit gets the foundry tests up to parity for the `AssetReceiver` and `Transactor` and then deletes the hardhat tests. Now this package only has foundry based tests which makes it much easier to standardize longer term with a single way of developing smart contracts in the monorepo.

### CODE REVIEW

File: packages/contracts-periphery/contracts/foundry-tests/AssetReceiver.t.sol [hunk 1]
```diff
@@ -17,6 +17,21 @@ contract AssetReceiver_Initializer is Test {
...
+        uint256 amount
+    );
+    event WithdrewERC721(
+        address indexed withdrawer,
+        address indexed recipient,
+        address indexed asset,
+        uint256 id
+    );
+
     function _setUp() public {
```
Comment (maurelian) [inline packages/contracts-periphery/contracts/foundry-tests/AssetReceiver.t.sol:35]: I think we should use this PLR as an opportunity to start moving away from this pattern.
I.e. if we make the following change, then we can delete the other `setUp` functions (which are just thin wrappers) from the child contracts: 
```suggestion
    function setUp() public {
```

**Review by maurelian — APPROVED** [review]
Some nice-to-have suggestions, but LGTM as is.

File: packages/contracts-periphery/contracts/foundry-tests/AssetReceiver.t.sol (hunk 1 above)
Comment (tynes) [reply packages/contracts-periphery/contracts/foundry-tests/AssetReceiver.t.sol]: I have cleaned this up, good call

File: packages/contracts-periphery/contracts/foundry-tests/AssetReceiver.t.sol [hunk 2]
```diff
@@ -52,6 +67,8 @@ contract AssetReceiverTest is AssetReceiver_Initializer {
         // Check that contract balance is 0 initially
         assertEq(address(assetReceiver).balance, 0);
 
+        vm.expectEmit(true, true, true, true);
```
[the remark this answers is not in the document]
Comment (tynes) [reply packages/contracts-periphery/contracts/foundry-tests/AssetReceiver.t.sol]: Addressed this, good call

**Comment by tynes** [comment]
I've addressed all comments here, should be good to merge pending passing tests

**Comment by tynes** [comment]
depcheck fail

```
Unused devDependencies
* @types/chai
* @types/mocha
* mocha

Exited with code exit status 255

```

**Pull request merged**